File: dlibmodule/src/main/java/au/hao/and/dlibmodule/Algorithms.java
package au.hao.and.dlibmodule;

import java.util.ArrayList;

public class Algorithms {
   public native double binomialrandomvarsaredifferent(String k1,String n1, String k2, String n2);
   public native double correlation(ArrayList<Integer> a,ArrayList<Integer> b);

    public double getDifference_binomial_random_vars(int k1,int n1,int k2, int n2){
        return binomialrandomvarsaredifferent(String.valueOf(k1),String.valueOf(n1),String.valueOf(k2),String.valueOf(n2));

    }
    private void initlib(){
        System.loadLibrary("algorithms");
    }

    public static Algorithms algorithms=null;

    public static Algorithms init(){
        if(algorithms==null){
            algorithms=new Algorithms();
            algorithms.initlib();
        }
        return algorithms;
    }
}

File: dlibmodule/src/main/java/au/hao/and/dlibmodule/Algorithms.kt
package au.hao.and.dlibmodule

import java.util.*
import kotlin.collections.ArrayList
import kotlin.experimental.and

class Algorithms {
    external fun binomialrandomvarsaredifferent(k1:String,n1:String,k2:String,n2:String):Double
    external fun correlation(a:ArrayList<Int>,b:ArrayList<Int>):Double

    fun getDifference_binomial_random_vars(k1:Int,n1:Int,k2:Int,n2:Int):Double{
        return binomialrandomvarsaredifferent(k1.toString(),n1.toString(),k2.toString(),n2.toString())
    }

    private fun initlib(){
        System.loadLibrary("algorithms")
    }
    companion object{
        var algorithms:Algorithms?=null

        fun init():Algorithms{
            if(algorithms==null)
            {
                algorithms= Algorithms()
                algorithms!!.initlib()
            }
            return algorithms!!
        }
    }
}